compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/ConfigMethodGen.java

```diff
@@ -150,13 +163,22 @@ private void generateInvokeConfiguration(MethodVisitor mv, PackageID id) {
         mv.visitInsn(ARRAYLENGTH);
         Label elseLabel = new Label();
         mv.visitJumpInsn(IFEQ, elseLabel);
-        mv.visitVarInsn(ALOAD, 3);
+        mv.visitVarInsn(ALOAD, 0);
         mv.visitFieldInsn(GETSTATIC, initClass, CURRENT_MODULE_VAR_NAME, GET_MODULE);
         mv.visitVarInsn(ALOAD, 4);
         mv.visitMethodInsn(INVOKESTATIC, LAUNCH_UTILS, "addModuleConfigData", ADD_MODULE_CONFIG_DATA, false);
         mv.visitLabel(elseLabel);
     }
 
+    private void generateInvokeConfigureInit(MethodVisitor mv, PackageID id) {
+        String configClass = JvmCodeGenUtil.getModuleLevelClassName(id, CONFIGURATION_CLASS_NAME);
+        mv.visitVarInsn(ALOAD, 0);
+        mv.visitVarInsn(ALOAD, 1);
+        mv.visitVarInsn(ALOAD, 2);
+        mv.visitVarInsn(ALOAD, 3);
```

Comment:
Since this change is from the dependent PR https://github.com/ballerina-platform/ballerina-lang/pull/42433, in an offline discussion with @heshanpadmasiri we decided to apply related code suggestions in a separate PR. It will be sent after this PR gets merged.